src/main/java/org/interledger/cryptoconditions/ConditionImpl.java

```diff
@@ -2,17 +2,29 @@
 
 import java.util.EnumSet;
 
+
 import org.interledger.cryptoconditions.Condition;
 import org.interledger.cryptoconditions.ConditionType;
 import org.interledger.cryptoconditions.FeatureSuite;
+import org.interledger.cryptoconditions.encoding.Base64Url;
 
-final class ConditionImpl implements Condition {
+public final class ConditionImpl implements Condition {
 
+	// Condition Interface related members 
 	private final ConditionType type;
 	private final EnumSet<FeatureSuite> features;
 	private final byte[] fingerprint;
 	private final int maxFulfillmentLength;
 
+	// URISerializable Interface related members
+	private static final String CONDITION_REGEX = "^cc:([1-9a-f][0-9a-f]{0,3}|0):[1-9a-f][0-9a-f]{0,15}:[a-zA-Z0-9_-]{0,86}:([1-9][0-9]{0,17}|0)$";
+	private static final java.util.regex.Pattern p = java.util.regex.Pattern.compile(CONDITION_REGEX);
+	
+	@SuppressWarnings("unused")
+	private ConditionImpl() {
```

Comment:
"Copy & Paste" from the five-ledger implementation. It just help to check that the input String, even before parsing, match the expected format. Makes code slightly slower but helps in debugging errors.